core/templates/dev/head/domain/objects/NumberWithUnitsObjectFactory.js

```diff
@@ -65,6 +65,22 @@ oppia.factory('NumberWithUnitsObjectFactory', [
       return numberWithUnitsString;
     };
 
+    NumberWithUnits.prototype.toMathjsCompatibleString = function() {
+      var numberWithUnitsString = '';
+      var unitsString = UnitsObjectFactory.fromList(this.units).toString();
```

Comment:
Consider calling `UnitsObjectFactory.fromList(this.units).toMathjsCompatibelString();` directly. Use toString function withing it to get unitString.